pkg/controller/controlplanecerts/controlplanecerts_controller_test.go

```diff
@@ -310,6 +357,15 @@ func (f *fakeClusterDeploymentWrapper) defaultCert(name, secretName string) *fak
 	return f
 }
 
+func (f *fakeClusterDeploymentWrapper) adminKubeconfig() *fakeClusterDeploymentWrapper {
+	f.cd.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
```

Comment:
May as well make this the default for the ClusterDeployment. You are already setting the ClusterDeployment to installed by default.

Athough, my preference would be that you use a mock remoteclientbuilder rather than writing a kubeconfig secret.
1. This will allow you to more easily test other unhappy scenarios such as a bad URL or errors from the remoteclientbuilder.
2. This will keep the code similar to the other controllers that use the mock remoteclientbuilder in their unit tests.